coralogix/resource_coralogix_apikeys.go

```diff
@@ -0,0 +1,359 @@
+)
+
+func NewApiKeyResource() resource.Resource {
+	return &ApiKeyResource{}
+}
+
+type ApiKeyResource struct {
+	client *clientset.ApikeysClient
+}
+
+func (r *ApiKeyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_apikeys"
+
+}
+
+func (r *ApiKeyResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	clientSet, ok := req.ProviderData.(*clientset.ClientSet)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *clientset.ClientSet, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = clientSet.ApiKeys()
+}
+
+func (r *ApiKeyResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (r *ApiKeyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: "ApiKey ID.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Api Key name.",
+			},
+			"value": schema.StringAttribute{
+				Computed:            true,
+				Sensitive:           true,
+				MarkdownDescription: "Api Key value.",
+			},
+			"owner": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"team_id": schema.Int64Attribute{
+						Optional: true,
+						Validators: []validator.Int64{
+							int64validator.ConflictsWith(
+								path.MatchRelative().AtParent().AtName("user_id"),
+							),
+						},
+					},
+					"user_id": schema.StringAttribute{
+						Optional: true,
+						Validators: []validator.String{
+							stringvalidator.ConflictsWith(
+								path.MatchRelative().AtParent().AtName("team_id"),
+							),
+						},
+					},
+				},
+				Required:            true,
+				MarkdownDescription: "Api Key Owner.It can either be a team_id or a user_id ",
+			},
+
+			"active": schema.BoolAttribute{
+				Required:            true,
+				MarkdownDescription: "Api Key Is Active.",
+			},
+			"hashed": schema.BoolAttribute{
+				Required:            true,
+				MarkdownDescription: "Api Key Is Hashed.",
+			},
+			"roles": schema.ListAttribute{
+				Required:            true,
+				ElementType:         types.StringType,
+				MarkdownDescription: "Api Key Roles",
+				Validators: []validator.List{
+					listvalidator.UniqueValues(),
+					listvalidator.ValueStringsAre(stringvalidator.OneOf("SCIM")),
```

Comment:
Not sure I understand the validations here. Also consider using `schema.SetAttribute` if the order doesn't matter.

Reply:
Not sure the best way here, but for now we are supporting creating api key with a single role called "SCIM"